### PR TITLE
Rewrite of base85.py algorithm

### PR DESCRIPTION
**Pull request by ChrisO345**
### Describe your change:

Fixes #8966
Fixes #8965

See #6216

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by rohan472000** [comment]
Now build will pass as #9078 is merged.

File: ciphers/base85.py [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
+"""
 
-def base85_encode(string: str) -> bytes:
+
+def _base10_to_85(d: int) -> str:
+    return "".join(chr(d % 85 + 33)) + _base10_to_85(d // 85) if d > 0 else ""
+
+
+def _base85_to_10(digits: list) -> int:
+    return sum(char * 85**i for i, char in enumerate(digits[::-1]))
```
Comment (tianyizheng02) [inline ciphers/base85.py:13]:
```suggestion
    return sum(char * 85**i for i, char in enumerate(reversed(digits)))
```

File: ciphers/base85.py [hunk 2]
```diff
@@ -1,30 +1,57 @@
...
-    >>> base85_encode("12345")
+    >>> ascii85_encode(b"12345")
     b'0etOA2#'
-    >>> base85_encode("base 85")
+    >>> ascii85_encode(b"base 85")
     b'@UX=h+?24'
     """
-    # encoded the input to a bytes-like object and then a85encode that
-    return base64.a85encode(string.encode("utf-8"))
+    binary_data = "".join([bin(ord(d))[2:].zfill(8) for d in data.decode("utf-8")])
```
Comment (tianyizheng02) [inline ciphers/base85.py:25]: Can we use a generator instead?
```suggestion
    binary_data = "".join(bin(ord(d))[2:].zfill(8) for d in data.decode("utf-8"))
```

File: ciphers/base85.py [hunk 3]
```diff
@@ -1,30 +1,57 @@
...
+    null_values = (32 * ((len(binary_data) // 32) + 1) - len(binary_data)) // 8
+    binary_data = binary_data.ljust(32 * ((len(binary_data) // 32) + 1), "0")
+    b85_chunks = [int(_s, 2) for _s in map("".join, zip(*[iter(binary_data)] * 32))]
+    result = "".join(
+        item for item in [_base10_to_85(chunk)[::-1] for chunk in b85_chunks]
+    )
```
Comment (tianyizheng02) [inline ciphers/base85.py:31]: Why not just
```suggestion
    result = "".join(
        _base10_to_85(chunk)[::-1] for chunk in b85_chunks
    )
```

File: ciphers/base85.py [hunk 4]
```diff
@@ -1,30 +1,57 @@
...
+    >>> ascii85_decode(b"@UX=h+?24")
+    b'base 85'
     """
-    # a85decode the input into bytes and decode that into a human readable string
-    return base64.a85decode(a85encoded).decode("utf-8")
+    null_values = 5 * ((len(data) // 5) + 1) - len(data)
+    binary_data = data.decode("utf-8") + "u" * null_values
+    b85_chunks = map("".join, zip(*[iter(binary_data)] * 5))
+    b85_segments = [[ord(_s) - 33 for _s in chunk] for chunk in b85_chunks]
+    results = [(bin(_base85_to_10(chunk))[2::].zfill(32)) for chunk in b85_segments]
```
Comment (tianyizheng02) [inline ciphers/base85.py:48]:
```suggestion
    results = [bin(_base85_to_10(chunk))[2::].zfill(32) for chunk in b85_segments]
```

File: ciphers/base85.py [hunk 5]
```diff
@@ -1,30 +1,57 @@
...
+    char_chunks = [
+        [chr(int(_s, 2)) for _s in map("".join, zip(*[iter(r)] * 8))] for r in results
+    ]
+    result = "".join("".join(char) for char in char_chunks)
+    offset = 0 if null_values % 5 != 0 else 1
```
Comment (tianyizheng02) [inline ciphers/base85.py:53]:
```suggestion
    offset = int(null_values % 5 == 0)
```

File: ciphers/base85.py [hunk 6]
```diff
@@ -1,30 +1,57 @@
...
+    return bytes(result[: -null_values + offset], "utf-8")
```
Comment (tianyizheng02) [inline ciphers/base85.py:54]:
```suggestion
    return bytes(result[: offset - null_values], "utf-8")
```

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: ciphers/base85.py (hunk 3 above)
Comment (ChrisO345) [inline ciphers/base85.py:31]:
```suggestion
    result = "".join(_base10_to_85(chunk)[::-1] for chunk in b85_chunks)
```

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**